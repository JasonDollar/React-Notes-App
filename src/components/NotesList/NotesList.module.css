.NotesList {
  /* flex: 3; */
  display: grid;
  grid-template-rows: repeat(auto-fit, minmax(10rem, 12rem));
  overflow: visible;
  /* max-height: 90vh; */
}
    @media (min-height: 90vh) {
      .NotesList {

        overflow-y: scroll;
      }
    }