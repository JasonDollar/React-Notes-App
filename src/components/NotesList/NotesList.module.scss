.NotesList {
  /* flex: 3; */
  // grid-area: "list";
  // flex: 1 1 100%;
  min-height: calc(100vh - 5rem);
  justify-self: stretch;
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  // grid-template-rows: auto 1fr;
  // max-height: 100vh;
  // overflow-y: scroll;
  // transition: all .3s;
  background: white;
  
  border-right: 1px solid #ebe8e8;
  // min-height: 100vh;
  // margin: 1rem 0;
  position: relative;

  @media (min-width: 576px) {
    padding-right: .5rem;
  }

}

.list {
  display: grid;
  // grid-template-rows: 10rem;
  // overflow-y: scroll;
  // flex: 1;
  list-style: none;
  margin: 0 1.5rem;
  // margin-top: 0;
  // padding: 0;
}

.paginationButtons {
  flex: 1;
  display: flex;
  justify-content: space-around;
  align-items: flex-end;
  // padding: 1rem 0;
  @media (max-width: 576px) {
    padding: 1.5rem 0;
  }
  & > * {
    flex: 0 1 45%;
  }
}