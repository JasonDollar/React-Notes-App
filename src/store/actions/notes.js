import * as actionTypes from './actionTypes'
import {db} from '../../firebase'


export const addNotetoStore = (note, id) => ({
  type: actionTypes.ADD_NOTE,
  payload: {
    title: note.title,
    body: note.body,
    createdAt: note.createdAt,
    editedAt: note.editedAt
  },
})

export const addNote = (note) => {
  return dispatch => {
    db.collection('notes').add(note)
      .then(docRef => dispatch(addNotetoStore(note, docRef.id)))
  }
}

export const removeNoteInStore = (id) => ({
  type: actionTypes.REMOVE_NOTE,
  payload: id,
})

export const removeNote = id => {
  return dispatch => {
    db.collection('notes').doc(id).delete()
      .then(() => dispatch(removeNoteInStore(id)))
  }
}

export const editNoteInStore = note => ({
  type: actionTypes.EDIT_NOTE,
  payload: {
    id: note.id,
    title: note.title,
    body: note.body,
    editedAt: note.editedAt
  }
})

export const editNote = (note) => {
  return dispatch => {
    db.collection('notes').doc(note.id).set(note)
      .then(() => dispatch(editNoteInStore(note))) 
  }
}

export const setNotesToStore = (notes) => ({
  type: actionTypes.SET_NOTES,
  payload: notes,
})

export const setNotes = () => {
  return dispatch => {
    const notes = []
     db.collection('notes').get()
      .then(querySnapshot => {
        querySnapshot.forEach(item => {
          const noteBody = item.data()
          // console.log(noteBody, item.id)
          notes.push({
            ...noteBody,
            id: item.id,
          })
        })
      })
      .then(() => dispatch(setNotesToStore(notes)))
  }
}