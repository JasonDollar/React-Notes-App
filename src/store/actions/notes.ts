import { Dispatch } from 'redux'
import * as actionTypes from './actionTypes'
import { ActionTypes } from './types'
import {firestore} from '../../data/firebase'
// import {history} from '../../data/history'

export interface Note {
  id?: string,
  title: string,
  body: string,
  createdAt?: number,
  editedAt: number,
  createdBy?: string
}

export interface AddNotetoStore {
  type: ActionTypes.ADD_NOTE,
  payload: Note
}

export interface RemoveNoteInStore {
  type: ActionTypes.REMOVE_NOTE,
  payload: string
}

export interface EditNoteInStore {
  type: ActionTypes.EDIT_NOTE,
  payload: Note
}

export interface SetNotesToStore {
  type: ActionTypes.SET_NOTES,
  payload: Note[]
}

export interface CleanNotes {
  type: ActionTypes.CLEAN_NOTES,
}

export const addNotetoStore = (note: Note, id: string): AddNotetoStore => ({
  type: ActionTypes.ADD_NOTE,
  payload: {
    id,
    title: note.title,
    body: note.body,
    createdAt: note.createdAt,
    editedAt: note.editedAt,
    createdBy: note.createdBy
  },
})

export const addNote = (note: Note) => {
  return (dispatch: Dispatch) => {
    return firestore.collection('notes').add(note)
      .then(docRef => {
        return dispatch(addNotetoStore(note, docRef.id))
        
        })
  }
}

export const removeNoteInStore = (id: string): RemoveNoteInStore => ({
  type: ActionTypes.REMOVE_NOTE,
  payload: id,
})

export const removeNote = (id: string) => {
  return (dispatch: Dispatch) => {
    return firestore.collection('notes').doc(id).delete()
      .then(() => dispatch(removeNoteInStore(id)))
  }
}

export const editNoteInStore = (note: Note): EditNoteInStore => ({
  type: ActionTypes.EDIT_NOTE,
  payload: {
    id: note.id,
    title: note.title,
    body: note.body,
    editedAt: note.editedAt
  }
})

export const editNote = (note: Note) => {
  return (dispatch: Dispatch) => {
    return firestore.collection('notes').doc(note.id).update(note)
      .then(() => dispatch(editNoteInStore(note))) 
  }
}

export const setNotesToStore = (notes: Note[]): SetNotesToStore => ({
  type: ActionTypes.SET_NOTES,
  payload: notes,
})

export const setNotes = (user: string) => {
  return (dispatch: Dispatch) => {
    const notes: any = []
    // const currentUser = firebase.auth().currentUser
     return firestore.collection('notes').where("createdBy", "==", user).get()
      .then(querySnapshot => {
        querySnapshot.forEach(item => {
          const noteBody = item.data()
          // console.log(noteBody, item.id)
          notes.push({
            ...noteBody,
            id: item.id,
          })
          return notes
        })
      })
      .then(() => dispatch(setNotesToStore(notes)))
  }
}

export const cleanNotes = (): CleanNotes => ({
  type: ActionTypes.CLEAN_NOTES
})