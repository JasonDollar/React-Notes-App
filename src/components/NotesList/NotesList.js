import React, {Component, useState} from 'react'
import {connect} from 'react-redux'
import uuid from 'uuid'
import classNames from 'classnames'

import {filterNotesInOrder} from '../../helpers'
import * as actions from '../../store/actions'
import NotesListItem from './NotesListItem/NotesListItem'
import NotesMenuActions from '../NotesMenuActions/NotesMenuActions'

import classes from './NotesList.module.scss'

const NotesList = ({notes, toggleNoteList, showNoteList}) => {

  const [filter, setFilter] = useState('')
  const [sortBy, setSortBy] = useState('createdAsc')
  const [activeClassId, setActiveClassId] = useState(0)

  const textFilterHandler = e => {
    const value = e.target.value
    setFilter(value)
  }
  const dropdownSortHandler = e => {
    const value = e.target.value
    console.log(value)
    setSortBy(value)
  }

  const containerClasses = classNames({
    [classes.NotesList]: true,
    [classes.active]: showNoteList,
  })
  const processedNotes = filterNotesInOrder(notes, filter, sortBy)
  
  return (
    <div className={containerClasses}>
      <NotesMenuActions setFilter={textFilterHandler} setSortBy={dropdownSortHandler} dropdownValue={sortBy}/>
      <ul className={classes.notesList}>
        {processedNotes.map(item => <NotesListItem note={item} key={item.id} toggleNoteList={toggleNoteList} isActive={activeClassId === item.id} setActiveClassId={setActiveClassId}/>)}
      </ul>
    </div>
  )
}


// class NotesList extends Component {
//   // componentDidMount = () => {
//   //   this.props.setNotes()
//   // }
  
//   render() {
//     const {notes} = this.props
//     return (
      
//     )
//   }
// }
const mapStateToProps = state => ({
  notes: state.notes
})

// const mapDispatchToProps = dispatch => ({
//   setNotes: () => dispatch(actions.setNotes())
// })

export default connect(mapStateToProps)(NotesList)
