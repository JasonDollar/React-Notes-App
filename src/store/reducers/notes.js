import * as actionTypes from '../actions/actionTypes'


const initialState = []

const notesReducer = (state = initialState, action) => {
  switch (action.type) {
    case actionTypes.ADD_NOTE: 
      return [
        ...state,
        action.payload
      ]
    case actionTypes.REMOVE_NOTE: 
      return state.filter(item => item.id !== action.payload)
    case actionTypes.EDIT_NOTE: 
      return state.map(item => {
        if (item.id === action.payload.id) {
          return {...action.payload, createdAt: item.createdAt}
        } else return item
      })
    case actionTypes.SET_NOTES: {
      return [
        ...action.payload
      ]
    }
    default: 
      return state
  }
}

export default notesReducer