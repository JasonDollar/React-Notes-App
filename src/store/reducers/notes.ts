import * as actionTypes from '../actions/actionTypes'
import { ActionTypes, NotesActions } from '../actions/types'
import { Note } from '../actions/notes'

const initialState: Note[] = []

const notesReducer = (state = initialState, action: NotesActions): Note[] => {
  switch (action.type) {
    case ActionTypes.ADD_NOTE: 
      return [
        ...state,
        action.payload
      ]
    case ActionTypes.REMOVE_NOTE: 
      return state.filter((item: Note) => item.id !== action.payload)
    case ActionTypes.EDIT_NOTE: 
      return state.map((item: Note) => {
        if (item.id === action.payload.id) {
          return {...action.payload, createdAt: item.createdAt}
        } else return item
      })
    case ActionTypes.SET_NOTES: {
      return [
        ...action.payload
      ]
    }
    case ActionTypes.CLEAN_NOTES: 
      return []
    default: 
      return state
  }
}

export { notesReducer }