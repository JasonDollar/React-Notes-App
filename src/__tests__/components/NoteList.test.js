import React from 'react'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'
import {notes} from '../../data/fixtures/notes'
import {NotesList} from '../../components/NotesList/NotesList'

let wrapper
beforeEach(() => {
  wrapper = shallow(<NotesList notes={notes} firebaseProcessing={false} isAuth />)
})

describe('NotesList component', () => {
  test('should match snapshot', () => {
    expect(toJSON(wrapper)).toMatchSnapshot()
  })
  test('should render list with proper length', () => {
    const elements = wrapper.find('NotesListItem')
    expect(elements.length).toBe(notes.length)
  })
})