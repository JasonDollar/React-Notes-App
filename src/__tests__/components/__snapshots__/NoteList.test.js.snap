// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotesList component should match snapshot 1`] = `
<styled.aside>
  <NotesMenuActions
    dropdownValue="createdAsc"
    setFilter={[Function]}
    setSortBy={[Function]}
  />
  <ul
    className="list"
  >
    <NotesListItem
      isActive={false}
      key="4sqNehvqUJO0cy69cDIk"
      note={
        Object {
          "body": "First note body",
          "createdAt": 1557247578961,
          "createdBy": "hxwM4zaAB5dmY9ifx6yx32ND5de2",
          "editedAt": 1557247578961,
          "id": "4sqNehvqUJO0cy69cDIk",
          "title": "First note",
        }
      }
    />
    <NotesListItem
      isActive={false}
      key="LEV5BAh2hwb9fh2kFepM2"
      note={
        Object {
          "body": "Second note body",
          "createdAt": 1557310812414,
          "createdBy": "hxwM4zaAB5dmY9ifx6yx32ND5de2",
          "editedAt": 1557310812414,
          "id": "LEV5BAh2hwb9fh2kFepM2",
          "title": "Second note",
        }
      }
    />
  </ul>
  <div
    className="paginationButtons"
  >
    <styled.button
      disabled={true}
      onClick={[Function]}
    >
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon="arrow-left"
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        symbol={false}
        title=""
        transform={null}
      />
       
      Previous
    </styled.button>
    <styled.button
      disabled={true}
      onClick={[Function]}
    >
      Next
       
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon="arrow-right"
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        symbol={false}
        title=""
        transform={null}
      />
    </styled.button>
  </div>
</styled.aside>
`;
